cli/src/main/java/org/jboss/sbomer/cli/feature/sbom/processor/DefaultProcessor.java

```diff
@@ -266,9 +275,102 @@ public Bom process(Bom bom) {
         // If there are any purl relcoations, process these.
         purlRelocations.forEach((oldPurl, newPurl) -> updatePurl(bom, oldPurl, newPurl));
 
+        addMissingNpmDependencies(bom);
```

Comment:
We do process only the main component? Shouldn't it be performed as part of the `processComponent` method to cover all potential cases?

Reply:
You mean to get all the transitive NPM dependencies for all the dependent components?  Is the Maven CycloneDX plugin traversing all the transitive dependency tree to create the component list? I don't remember if that is the case. If yes then I agree we should probably move `addMissingNpmDependencies` to all the components and not only the main one. Otherwise I believe it's fine?

Reply:
I think with my previous comment I wanted to touch on two things:

1. We probably should process for additional NPMs only components which are of type other than RPM and container images.
2. According to the code `addMissingNpmDependencies(Bom bom)` method is covering only the component located in the under the `.metadata.component`. Why aren't we looking at every component in `.components`? I imagine that the main use case that trigerred this RFE is the manifestation of a ZIP deliverable. Why would we want to not do it for any (Java?) builds?

What would the processed manifest look if we would scan the whole component list? I think for a ZIP it should be the same, correct?

Reply:
@vibe13 I guess you are better at answering this. I was going according to what I understood from your "1 PNC build == 1 SBOM"

Reply:
Ok, now I've read the comments. I was doing it with the understanding this is for the analysis of single PNC build. 

I do not fully understand what is the problem with the manifestation of the zip deliverable and how the deliverable analysis is turned into manifest.

Reply:
I think that for the single PNC builds we are fine with this PR. 
For the ZIP manifestation (see `CycloneDxGenerateOperationCommand`) we will need to hook in this logic to retrieve the NPM dependencies for all the components matched with PNC builds. I think I am fine with another PR for that! :-)